Client/play/static/player.app.js

```diff
@@ -175,18 +175,52 @@ var Status = React.createClass({displayName: "Status",
     return {
       playerAtLocation: false,
       huntComplete: false,
-      distanceToNextPin: 0.00
-    };
-  },  
+      distanceToNextPin: 0.00,
+      currentPin: null
+    }
+  },
+
+  getHuntStatus: function() {
+    var currentPin = Number.parseInt(this.props.hunt.get('currentPin'));   
+  },
+
+  getCurrentPin: function() {
+    var currentPin = Number.parseInt(this.props.hunt.get('currentPin'));
+    this.setState({currentPin: currentPin})
+     
+  },
+
+  visitedPins: [],  
 
   componentWillMount: function() {
     gMap.getDistanceByLocation(function (value) {
       var playerAtLocation = false;
-      if(value < 0.26){
+      
+      var currentPin = Number.parseInt(this.props.hunt.get('currentPin'));
```

Comment:
currentPin is recalculated several times. Would it make sense to calculate it once and store it in State?